### DISCUSSION TITLE
Show Artist and Title Question

### DISCUSSION THREAD

**Discussion by inovaxt** [schreibfaul1/ESP32-audioI2S#351] [Q&A] [answered]
It's a wonderful library.  Thank you @schreibfaul1 for the effort.  Could it be possible for you to show me how to grab **artist** and **title** only from void audio_id3data(const char *info)?    

That prints a lot of ID3 tag information, such as Track, PartOfSet, Title, Album, ContentType, Year, Artist, Band and  Composer, I only need artist and title only.

Thanks for any help, I appreciate it.

**Comment by schreibfaul1** [comment, marked as answer]
Hello inovaxt,
the ID3 tags are all evaluated if they are present. In the routine showID3Tag you can deactivate the tags that are of no interest to you. There are different tags in the ID3 versions, with V2.2 only rarely being found

**Comment by inovaxt** [comment]
Thank you @schreibfaul1 for the reply, I will study the showID3Tag & deactivate the tags and how to concatenate char* and char* into a char*.

**Comment by schreibfaul1** [comment]
this simple code works as long as there is an artist and title tag
````c++
void audio_id3data(const char *info){
    String tmp = info;
    static String artist = "";
    static String title = "";
    if(tmp.startsWith("Artist:")) artist = tmp.substring(8);
    if(tmp.startsWith("Title:")) title = tmp.substring(7);
    if(artist.length() > 0 && title.length() > 0){
        tmp = artist + " - " + title;
        Serial.println(tmp);
        artist = "";
        title = "";
    }
}
````

**Reply by inovaxt** [reply]
Once again, thank you so much for your great help, I appreciate it, I will test it and then get it back to you soon.
Thanks a lot.

**Reply by inovaxt** [reply]
> this simple code works as long as there is an artist and title tag
> 
> ```c++
> void audio_id3data(const char *info){
>     String tmp = info;
>     static String artist = "";
>     static String title = "";
>     if(tmp.startsWith("Artist:")) artist = tmp.substring(8);
>     if(tmp.startsWith("Title:")) title = tmp.substring(7);
>     if(artist.length() > 0 && title.length() > 0){
>         tmp = artist + " - " + title;
>         Serial.println(tmp);
>         artist = "";
>         title = "";
>     }
> }
> ```

Hi @schreibfaul1,

I ran below audio_showstreamtitle(const char *streamtitle) & audio_id3data(const char *info) without any issue, but I followed your code to catch Artist - Title, it popped-up the error "'class String' has no member named 'update'", I am googling that, so far I did not find the root cause yet, could you please kindly point me what the root cause is?

Thank you.

```c++
void audio_showstreamtitle(const char *streamtitle) {
  Serial.print("Stream Info: "); Serial.println(streamtitle);
  title.update(streamtitle);  // No issue
  dashboard.sendUpdates();
}
void audio_id3data(const char *info) {
  Serial.print("Info: "); Serial.println(info);
    title.update(info);  // No issue
    dashboard.sendUpdates();
}
```
=============below sketch occurred error=====================
```c++
void audio_id3data(const char *info) {
  //Serial.print("Info: "); //Serial.println(info);
  String tmp = info;
  static String artist = "";
  static String title = "";
  if (tmp.startsWith("Artist:")) artist = tmp.substring(8);
  if (tmp.startsWith("Title:")) title = tmp.substring(7);
  if (artist.length() > 0 && title.length() > 0) {
    tmp = artist + " - " + title;
    Serial.println(tmp);

    title.update(tmp);  // <-- error: 'class String' has no member named 'update'
    dashboard.sendUpdates();

    artist = "";
    title = "";
  }
}
```
[https://github.com/ayushsharma82/ESP-DASH](url)

**Reply by schreibfaul1** [reply]
if you want the example below to be the same as the example above try
title.update(tmp.c_str());

**Reply by inovaxt** [reply]
Hi @schreibfaul1,

Still no joy, same error "error: 'class String' has no member named 'update'".  Could you please kindly help to identify what could be the issue?

Thanks
